lib/src/main/java/com/auth0/jwt/algorithms/RSAAlgorithm.java

```diff
@@ -55,8 +62,65 @@
         this(new CryptoHelper(), id, algorithm, keyProvider);
     }
 
+    @Override
+    public void verifyWithX509(DecodedJWT jwt, EncodeType encodeType, String jwksFile, String pemFile) throws Exception {
```

Comment:
Can you write a wrapper for this that infers its EncodingType? That will probably work better in most cases

Reply:
good point.  i can have a constant that stores what type of encodingtype was used to sign the JWT with and use it accordingly.

Reply:
fixed locally

Reply:
Cool. Can you do anything for unknown JWTs? Can't we see what base something is in based off what characters are used?

Reply:
i dont think there's any other way.  i tried this, which keeps returning UTF-8: https://code.google.com/archive/p/juniversalchardet/

what i can try to do is run the decode() method for each of the base encodings for the particular string and whichever one doesnt throw an exception is the one.  is that what you want?